generics/helpers/elasticSearch.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * name : elasticSearch.js
+ * author : Aman Jung Karki
+ * created-date : 11-Jun-2020
+ * Description : elastic search common functionality
+ */
+
+/**
+  * Get user data
+  * @function
+  * @name get
+  * @param {Object} data
+  * @param {String} data.id - user id
+  * @param {String} data.index - elastic search index
+  * @param {String} data.type - elastic search type
+  * @returns {Object} Get user data.
+*/
+
+var get = function (data) {
+
+  return new Promise(async function (resolve, reject) {
+    try {
+
+      if (!data.id) {
+        throw "id is required";
+      }
+
+      if (!data.index) {
+        throw "index is required";
+      }
+
+      if (!data.type) {
+        throw "type is required";
+      }
+
+      const result = await elasticsearch.client.get({
+        id: data.id,
+        index: data.index,
+        type: data.type
+      }, {
+          ignore: [httpStatusCode["not_found"].status],
+          maxRetries: 3
+        });
+
+      return resolve(result);
+
+    } catch (error) {
+      return reject(error);
+    }
+  })
+};
+
+/**
+  * Create or update based on user data.
+  * @function
+  * @name createOrUpdate
+  * @param {Object} data - Data to be created or updated.
+  * @param {String} data.id - user id.
```

Comment:
@aman-tunerlabs - It could be any id right, I mean in general this function, it's not just for users so, the description can be **"a unique document id"**